reward/reward_distributor.go

```diff
@@ -38,113 +50,456 @@ type governanceHelper interface {
 	ParamsAt(num uint64) (*params.GovParamSet, error)
 }
 
-type RewardDistributor struct {
-	rcc *rewardConfigCache
-	gh  governanceHelper
+type rewardConfig struct {
+	// hardfork rules
+	rules params.Rules
+
+	// values calculated from block header
+	totalFee *big.Int
+
+	// values from ChainConfig
+	mintingAmount *big.Int
+	minimumStake  *big.Int
+	deferredTxFee bool
+
+	// parsed ratio
+	cnRatio    *big.Int
+	kgfRatio   *big.Int
+	kirRatio   *big.Int
+	totalRatio *big.Int
+
+	// parsed KIP82 ratio
+	cnProposerRatio *big.Int
+	cnStakingRatio  *big.Int
+	cnTotalRatio    *big.Int
+}
+
+type RewardSpec struct {
+	Minted   *big.Int                    // the amount newly minted
+	TotalFee *big.Int                    // total tx fee spent
+	BurntFee *big.Int                    // the amount burnt
+	Proposer *big.Int                    // the amount allocated to the block proposer
+	Stakers  *big.Int                    // total amount allocated to stakers
```

Comment:
Proposer is singular and Stakers are plural. It represents some actor to me. How about appending Amount string?

Reply:
To do this, we'll need to change [kip82](https://github.com/klaytn/kips/pull/82/files#diff-a519db6b0fc4c2a08964baca4b2093b909f54cd942ba6dfe6eea9d306a2e4367R115). I don't think it'll be possible in this PR

Reply:
I have looked at kip82 also quickly. If some one read kip82 in advance, they understand stakers meaning. I agree your opinion.